### PR TITLE
Move MSFT stuff to separate header file

### PR DESCRIPTION
**Pull request by richsalz**
This cuts the size of e_os.h in half for non-Microsoft platforms.

### CODE REVIEW

File: e_os.h [hunk 1]
```diff
@@ -94,149 +88,13 @@ extern "C" {
...
-#  ifndef R_OK
-#   define R_OK        4
-#  endif
-#  ifdef OPENSSL_SYS_WINCE
-#   define DEFAULT_HOME  ""
-#  else
-#   define DEFAULT_HOME  "C:"
-#  endif
+# if defined(WIN32) || defined(WINDOWS) || defined(MSDOS)
+#  include "include/internal/plat-msft.h"
```
Comment (levitte) [inline e_os.h]: You do know that we have a os-dep directory for this kind of thing, right?

Comment (richsalz) [reply e_os.h]: Yes, but this seems different.  Should e_os include os-dep files?  Looking at the one file that's there, it seems "special" purpose.  But I am willing to change if you think so.

Comment (levitte) [reply e_os.h]: The haiku.h file is included through compiler flags, it's true, but I think it's kind of a special case (unless such practice is common among compilers these days).

We have talked long ago about having an OS dependent set of source files, we just haven't really started that work yet, except for the haiku header file.  So I'd say that using that same space for more OS dependent stuff, and I see no real reason why `e_os.h` couldn't be used to include them.  Ultimately, `e_os.h` could become a bunch of conditional includes of files in `os-dep`...

So yes, I think so.

**Comment by richsalz** [comment]
Moved the file. additional commit (to be squashed) pushed.

File: include/internal/plat-msft.h [hunk 2]
```diff
@@ -0,0 +1,166 @@
+/*
```
Comment (levitte) [inline include/internal/plat-msft.h]: I think you can remove this one now ;-)

Comment (richsalz) [reply include/internal/plat-msft.h]: GitHub confusion?  I'll do another push/-f but I did 'git mv' the file.

**Comment by dot-asm** [comment]
I'd like to have some time to ponder about this...

**Comment by richsalz** [comment]
Take all the time you need, @dot-asm.

**Comment by richsalz** [comment]
Ping, anyone?  Makes e_os.h much simpler for everyone.

**Comment by levitte** [comment]
I'm wondering if we shouldn't divide it further.  That would really make life simpler.

Also, is it important to have this in pre-beta?

**Comment by richsalz** [comment]
Maybe it can make sense to split it further, if someone wants to do that in another PR.  It is hard to see this as a bug-fix -- how would you write the bug? -- which is why I want it now.

**Comment by levitte** [comment]
Well, I would argue that it doesn't add anything new, it just moves things around for clarity.

As for a report...  "e_os.h hard to read, can it be made more clear?"  There you go.

**Comment by dot-asm** [comment]
Problem with mechanical split such as suggested is that it doesn't really address the pending problems, but simply make them prettier, problems that is. For starters I'd say that "microsoft" is Windows-only. For example DJGPP stuff is totally thing of its own. [In the essence it resembles "hosted" kind of platform on Windows, it's kind of a POSIX-y layer,  and it's probably more appropriate to treat it as such.] Another problem is that one wishes to unify all that "underscoring" such as #define stat _stat spread across the source. Problem here is that one needs it in apps (though not for same reason as in lib), but one doesn't want apps to depend on [overly] private header such as e_os.h (or its mechanical split). And finally, and maybe most importantly, it's desirable to arrange things so that <winsock2.h> and <windows.h> are included only when actually needed.

**Comment by richsalz** [comment]
Those are all good suggestions for further improvements.  I'm just trying to move all MSFT (and its dependencies) stuff somewhere else. :)  Yes, this is purely cosmetic.  A big cleanup of e_os is beyond this release.

**Comment by richsalz** [comment]
The question is: are things better with this PR?  Does it help get to a more clean world?

**Comment by dot-asm** [comment]
With #5676 in mind. I for one am not against occasional inclusion of <windows.h> explicitly (so that one can see upfront that some specific file contains real(*) Win32 code), but it does have its cons. For example I'd appreciate one place where one can specify minimally supported _WIN32_WINNT. And trouble is that it has to be defined prior <windows.h>. And not to forget dependency on order of <winsock2.h> and <windows.h> inclusion. (Well, one can include them in reverse order, but then you'd be missing some new winsock2 things that we want.) Just in case, in #5676 it's exclusively about CP_UTF8, and it's available for all Windows, so there is no strong dependency in _WIN32_WINNT. Would we consider it acceptable to occasionally include <windows.h> explicitly, in cases of weak dependency on _WIN32_WINNT?

(*) "real Win32" effectively refers to functions that reside in KERNEL32.DLL, ADVAPI32.DLL and alike. I.e. not ANSI C and POSIX-looking things like _stat that are implemented in VCRT.

**Comment by levitte** [comment]
If all the windows code needs `e_os.h` is to get windows.h and winsock2.h, I see no reason not to include them directly.

Also, I have no idea how `_WIN32_WINNT` works exactly, but if it's anything like, for example, defining `_POSIX_C_SOURCE` to tell the system headers what features are desired, then I see no reason why it can't be defined directly in the source.  If it's more complex than that, then I can see reason not to have to repeat ourselves all over the place...

**Comment by dot-asm** [comment]
And the fact that #5676 was required also serves as an example of danger of mechanical pretty-fication of headers. I mean they should be taken in broader context, with greater care and cross-platform consideration... I mean problem is that you say "what can go wrong," and one simply can't answer affirmatively "nothing"...

**Comment by dot-asm** [comment]
> If all the windows code needs e_os.h is to get windows.h and winsock2.h, I see no reason not to include them directly.

Well, it's a bit of a trick question. e_os.h has even some _MSC_VER-specific fixups, so it's not exclusively about <windows.h>. And besides, there is ambiguity in what "windows code" actually entails. As implied is footnote in one of my previous remarks, "real Win32" code does need <windows.h>, but we do have #ifdef _WIN32 clauses that don't require <windows.h>...

**Comment by richsalz** [comment]
Closing this.  Perhaps someone will do the "right thing" for e_os.h later on.

**Pull request closed without merging**